sky/clouds/aws.py

```diff
@@ -350,16 +349,21 @@ def _make(instance_list):
                     # Setting this to None as AWS doesn't separately bill /
                     # attach the accelerators.  Billed as part of the VM type.
                     accelerators=None,
+                    cpu=None,
                 )
                 resource_list.append(r)
             return resource_list
 
         # Currently, handle a filter on accelerators only.
         accelerators = resources.accelerators
         if accelerators is None:
-            # No requirements to filter, so just return a default VM type.
-            return (_make([AWS.get_default_instance_type()]),
-                    fuzzy_candidate_list)
+            # Return a default instance type with the given number of vCPUs.
+            default_instance_type = AWS.get_default_instance_type(
```

Comment:
For discussion/a thought: This is now dynamically picking a VM type based on `cpu`, which means it's no longer a "default" constant. More like a `service_catalog.get_instance_type_for_cpu()` a la L371.  

Should we remove this interface (Cloud.get_default_instance_type()) then? Wdyt? (I can see arguments for both choices.)

Reply:
I think it cannot be `get_instance_type_for_cpu()` because it limits the search space to the **default** instance family and uses 8 as the **default** number of the vCPUs. I think one can easily understand what it means.

Reply:
The naming part sounds good. Sorry, I did not spell out in my comment but was thinking removing one interface from Cloud will make it easier to add new clouds. Also, it's a shallow method (1-line dispatch). How about removing it and just calling `service_catalog.get_default_instance_type()` directly here?

Reply:
Wouldn't it be still useful to explicitly show what the default type is?

Reply:
Do you mind clarifying what you meant? 

Currently, the Cloud interface includes a shallow method, which does not show the default type:
```python
    @classmethod
    def get_default_instance_type(cls,
                                  cpu: Optional[str] = None) -> Optional[str]:
        return service_catalog.get_default_instance_type(cpu=cpu, clouds='aws')
```

Reply:
I mean one can easily find which instance type is default for AWS by calling `AWS.get_default_instance_type`, which is good.

Reply:
Makes sense.